### ISSUE TITLE
For LSTMTimeStep, is there any difference between training multiple pieces of data at one time and training one piece of data multiple times?

### ISSUE THREAD

**Issue by aweiu** [BrainJS/brain.js#838]
```
const net = new brain.recurrent.LSTMTimeStep();

// train multiple pieces of data at one time
net.train([
  { input: [1, 3], output: [0] },
  { input: [2, 4], output: [1] },
  { input: [5, 7], output: [0] },
])

// train one piece of data multiple times
net.train([
  { input: [1, 3], output: [0] },
])
net.train([
  { input: [2, 4], output: [1] },
])
net.train([
  { input: [5, 7], output: [0] },
])
```
According to my own test, they are different, so how can I determine how many pieces of data to train at a time?

**Comment by robertleeplummerjr** [comment]
Yes, it is slightly different.

Generally the best practice is to train on a larger number of pieces rather than a smaller.  That isn't always the case, but a general rule of thumb.  

The reason is the iterations on the training data.  Say iff we train 20k times on a single data item, the network will very likely overtrain.  If you have multiple items, then it is less likely.

There is also the danger of training too much on the data you have, which can overtrain or result in vanishing or exploding weights/gradients.

The best option would probably be to use a subset of data that is relevant, or something like https://github.com/BrainJS/brain.js#cross-validation.

**Comment by aweiu** [comment]
@robertleeplummerjr Thank you for your reply. I have another related question

Assuming that I have completed the training, I found that I can use a single runData or multiple runData: 

```
// single runData 
net.run([
  [1, 3]
])

// multiple runData
net.run([
  [1, 3],
  [2, 4]
])
```

but I don't know:

1. Which one is better?
2. If it is better to use multiple rundata, how many should I use?
